js/utils.js

```diff
@@ -961,6 +965,16 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
           _.set(data, ['record', 'data', data.field.column], '');
         }
 
+        if (data.field.from === 'details') {
+          if (!Array.isArray(data.record.data[data.field.column])) {
+            data.record.data[data.field.column] = [];
```

Comment:
Given I'm reading this right, if the data is not an array, you're turning it into an empty array. What if there was a value but it's not an array. Do you end up wiping the value and replacing it with an empty array? Does this not have any repercussion in losing the data?

Reply:
Yes, you are right it may lead us to lose the data.
But there is another issue.
Without this change, when users are setting the type to `image` and selecting the data type `image folder` the record in this `data.record.data[data.field.column]` object stays empty and we can't display images that in the folder.
Is there any way we could add images from the folder to this object `data.record.data[data.field.column]` without this change?

Reply:
I still don't understand why it's a problem for `data.record.data[data.field.column]` to stay exactly as it is, which could be a string, number, undefined or anything else.

I also see a few lines up that there's a section of code like this:

```js
        var image = _.get(data, ['record', 'data', data.field.column]);

        if (_.isArray(image)) {
          image = _.compact(image)[0];
        }

        if (isValidImageUrl(image)) {
          // Record data doesn't need updating
          return data.record;
        }
```

This suggests that there are scenarios where an array of images will be changed to only show the first one. That doesn't sound like this meets our requirements.

Reply:
@tonytlwu I've made some [screencasts](https://share.getcloudapp.com/o0uEQ895) with voice over where I'm trying to explain why I've done those changes and the code above. I hope it will make something clearer.